src/app/core/signing.guard.ts

```diff
@@ -49,10 +49,14 @@ export class SigningGuard implements CanActivate {
         return false;
       }
 
-      if (this.msgIOService.checkForMessage(SigningGuard.uid)) {
-        this.router.navigate(['/rejecc']);
-        return false;
-      }
+      this.msgIOService.checkForMessage(SigningGuard.uid).then((signed) => {
+        if (signed === true) {
```

Comment:
Since I've strongly typed checkForMessage to only ever return a boolean, the strict equality operator isn't necessary since there never should be a non-`boolean` getting put into the condition. Therefore, you can use the shorthand here:
```suggestion
        if (signed) {
```